appliances/Harbor/appliance.sh

```diff
@@ -0,0 +1,325 @@
+    download_unpack_harbor
+    msg info "Installation phase finished"
+}
+
+### Configuration Stage => Configures Harbor YAML and SSL certificates
+service_configure() { 
+    msg info "Starting configuration..."
+    mount_persistent
+
+    ## SSL CERTS
+
+    if [ ! -d "/root/certs" ]; then
+        mkdir -p /root/certs
+    fi
+
+    # If one of both files (cert or key) does not exist...
+    if [ ! -f "/root/certs/server.crt" ] || [ ! -f "/root/certs/server.key" ]; then
+        if [ "$HARBOR_SSL_CERT" = "AG" ] || [ "$HARBOR_SSL_KEY" = "AG" ]; then
+            msg info "Autogenerating SSL certificates..."
+            generate_ssl_certs
+        else
+            msg info "Configuring provided SSL certificates..."
+            echo $HARBOR_SSL_CERT >> mkdir /root/certs/server.crt
+            echo $HARBOR_SSL_KEY >> mkdir /root/certs/server.key
+        fi
+    else
+        msg info "The certificates already exist"
+        msg info "Recontextualization detected. This appliance is not prepared for reconfiguration. Skipping..."
+        return 1
+    fi
+
+    msg info "Configuring Harbor YAML file..."
+
+    gawk -i inplace -v cert="/root/certs/server.crt" -v key="/root/certs/server.key" -v hostname="$HARBOR_HOSTNAME" '
+    {
+        if ($1 == "certificate:") {
+            print "  certificate:", cert
+        } else if ($1 == "private_key:") {
+            print "  private_key:", key
+        } else if ($1 == "hostname:") {
+            print "hostname:", hostname
+        } else {
+            print
+        }
+    }' /root/harbor/harbor.yml
+
+    gawk -i inplace -v admin_pwd="$HARBOR_ADMIN_PASSWORD" -v db_pwd="$HARBOR_DB_PASSWORD" '
+    {
+        if ($1 == "harbor_admin_password:") {
+            print "harbor_admin_password:", admin_pwd
+        } else if ($1 == "password:") {
+            print "  password:", db_pwd
+        } else {
+            print
+        }
+    }' /root/harbor/harbor.yml
+    msg info "Configuration phase finished"
+}
+
+service_bootstrap() { # Running Harbor installer => will generate and run docker-compose
+    msg info "Starting bootstrap..."
+    /root/harbor/install.sh
+    if [ $? -ne 0 ]; then
+	    msg error "Harbor installation script failed, aborting..."
+        exit 1
+    else
+        msg info "Harbor installation script finished successfully. Waiting 60s before final health checks..."
+    fi
+    sleep 60
+    msg info "Running final health checks..."
+    wait_for_docker_containers
+    cleanup_installation
+    msg info "Bootstrap phase finished"
+}
+
+### Function Definitions
+
+check_internet_access() {
+    # Ping Google's public DNS server
+    if ping -c 1 8.8.8.8 &> /dev/null; then
+        msg info "Internet access OK"
+        return 0
+    else
+        msg error "The VM does not have internet access. Aborting Harbor deployment..."
+        exit 1
+    fi
+}
+
+install_requirements(){
+    apt update && apt install openssl ca-certificates curl gnupg docker.io docker-compose -y
```

Comment:
I think in scripts `apt-get` should be preferred together with `export DEBIAN_FRONTEND=noninteractive`  :thinking:

Reply:
Fixed in https://github.com/OpenNebula/one-apps/pull/87/commits/7745f4ab91ecbd2a9bc3632d4ec3bf18ea8c4003